### PR TITLE
Move SegmentFeatures base class into complex

### PR DESCRIPTION
**Pull request by imikejackson**
This was done so that other filters can use this base class when they are stored in other plugins

### CODE REVIEW

File: src/Plugins/ComplexCore/test/RotateSampleRefFrameTest.cpp [hunk 1]
```diff
@@ -1,219 +0,0 @@
-#include <catch2/catch.hpp>
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/RotateSampleRefFrameTest.cpp:1]: Is this supposed to be here?

**Review by jmarquisbq — CHANGES_REQUESTED** [review]

File: src/Plugins/ComplexCore/test/RotateSampleRefFrameTest.cpp (hunk 1 above)
Comment (imikejackson) [reply src/Plugins/ComplexCore/test/RotateSampleRefFrameTest.cpp:1]: The file has been completely removed.

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/ScalarSegmentFeaturesFilter.cpp [hunk 2]
```diff
@@ -151,9 +154,12 @@ IFilter::PreflightResult ScalarSegmentFeaturesFilter::preflightImpl(const DataSt
     dataPaths.push_back(goodVoxelsPath);
   }
 
-  data.validateNumberOfTuples(dataPaths);
+  if(!dataStructure.validateNumberOfTuples(dataPaths))
+  {
+    return {nonstd::make_unexpected(std::vector<Error>{Error{-651, fmt::format("Input arrays do have matching tuple counts.")}})};
```
[the remark this answers is not in the document]
Comment (imikejackson) [reply src/Plugins/ComplexCore/src/ComplexCore/Filters/ScalarSegmentFeaturesFilter.cpp]: Opps. Fixed in latest push.

**Comment by JDuffeyBQ** [comment]
If moving `RotateSampleRefFrame` is intended as part of this PR it should be documented.

**Pull request merged**